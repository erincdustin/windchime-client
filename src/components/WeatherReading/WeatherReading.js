import React from 'react';
import TokenService from '../../services/token-service';

function WeatherReading(props) {
  let imageUrl = '';

  let weather = {
    main: {
      temp: ''
    },
    weather: [{
      icon: '',
      id: '',
      main: '',
      description: ''
    }]
  }
  let weatherResults = '';

  if(props.weather !== null) {
    weather = props.weather;
    imageUrl = `https://openweathermap.org/img/wn/${props.weather.weather[0].icon}@2x.png`
    
    weatherResults=  
        <div className="ribbon one">
          <div className="ribbon-header">Weather</div>
          <span className="ribbon-text">
          <span className="hidden">{Math.floor(weather.main.temp)}&deg;</span>
          <button className="btn weather" onClick={() => {
            TokenService.clearWeatherToken();
            props.history.push('/getWeather')
          }}>Change</button>
          </span>
          <img className="icon" src={imageUrl} alt="weather icon"></img>
        </div>
  }

  return (
    <div>
      <div>{weatherResults}</div>
    </div>
  );
}

export default WeatherReading;