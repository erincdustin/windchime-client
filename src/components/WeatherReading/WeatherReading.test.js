import React from 'react';
import renderer from 'react-test-renderer';
import { library } from '@fortawesome/fontawesome-svg-core'
import { faWind } from '@fortawesome/free-solid-svg-icons'
import WeatherReading from './WeatherReading';

library.add(faWind)

describe(`Weather Reading`, () => {
  const props = {
    weather: {
      "coord": {
          "lon": -82.37,
          "lat": 27.09
      },
      "weather": [
          {
              "id": 500,
              "main": "Rain",
              "description": "light rain",
              "icon": "10n"
          }
      ],
      "base": "stations",
      "main": {
          "temp": 79.41,
          "pressure": 1016,
          "humidity": 83,
          "temp_min": 75.99,
          "temp_max": 82.99
      },
      "visibility": 16093,
      "wind": {
          "speed": 5.82,
          "deg": 120
      },
      "clouds": {
          "all": 20
      },
      "dt": 1562821257,
      "sys": {
          "type": 1,
          "id": 6184,
          "message": 0.0084,
          "country": "US",
          "sunrise": 1562841758,
          "sunset": 1562891211
      },
      "timezone": -14400,
      "id": 0,
      "name": "Venice",
      "cod": 200
  }
  }

  it('renders as expected', () => {
    const tree = renderer
      .create(<WeatherReading
            weather={props.weather}
            />)
      .toJSON();
    expect(tree).toMatchSnapshot();
  });
})