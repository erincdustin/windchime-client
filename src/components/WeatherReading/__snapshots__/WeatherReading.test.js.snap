// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Weather Reading renders as expected 1`] = `
<div>
  <div>
    <div
      className="ribbon one"
    >
      <div
        className="ribbon-header"
      >
        Weather
      </div>
      <span
        className="ribbon-text"
      >
        <span
          className="hidden"
        >
          79
          Â°
        </span>
        <button
          class="btn weather"
          onClick={[Function]}
        >
          Change
        </button>
      </span>
      <img
        alt="weather icon"
        className="icon"
        src="http://openweathermap.org/img/wn/10n@2x.png"
      />
    </div>
  </div>
</div>
`;
